controller/control.go

```diff
@@ -1035,3 +1038,14 @@ func (c *Controller) monitoring(address string, backend types.Backend) {
 	logrus.Infof("Monitoring stopped %v", address)
 	c.RemoveReplicaNoLock(address)
 }
+
+func (c *Controller) monitorPreload(address string) {
```

Comment:
why are we not using any locks in this function.

Reply:
have added in the recent commit